### DISCUSSION TITLE
Disable validation for whole form.

### DISCUSSION THREAD

**Discussion by azad-steadybiz** [react-hook-form/react-hook-form#2849] [Ideas]
There should be some argument in useForm by which we can disable validation for the whole form.

 Suppose I have put the validation for the whole form using RHF. Now for the some reason I have to remove the validation for a moment in whole form, so now what I do, comment all the validation part in each input field. So I am saying there should be some argument in useForm for disabling the validation in the whole form for a moment.

example:
const {} = useForm({ disableValidation:true })

**Comment by bluebill1049** [comment]
This is going to be very tricky because of all those external controlled inputs (MUI, antD and etc...) it's better at your app level to disable that submit button.

**Reply by azad-steadybiz** [reply]
No, In some cases after applying validation suppose I want to disable it for a moment by passing some argument in useFrom. This is what I talking about.

**Comment by PluviaCon** [comment]
Are you saying that when you submit the form, you don't validate the required options in the form?

**Reply by azad-steadybiz** [reply]
No, Suppose I have put the validation for the whole form using RHF. Now for the some reason I have to remove the validation for a moment in whole form, so now what I do,  comment all the validation part in each input field. So I am saying there should be some argument in useForm for disabling the validation in the whole form for a moment.

example:
const {} = useForm({ disableValidation:true })

**Reply by PluviaCon** [reply]
😂That's exactly what I need.

**Reply by azad-steadybiz** [reply]
That's great man.

**Reply by PluviaCon** [reply]
I was wondering if I could trigger it manually first, then clear the reset error.
The two methods `triggerValidation` & `clearError` implement

**Reply by bluebill1049** [reply]
if you are using resolver you can achieve that with context, or set all built-in validation with an extra state to control the validation.

**Reply by azad-steadybiz** [reply]
> if you are using resolver you can achieve that with context, or set all built-in validation with an extra state to control the validation.
Can you share example

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/lively-haze-nyhlc?file=/src/App.js

**Reply by PluviaCon** [reply]
😂I think it's easier to configure the properties.Like this:
![选区_011](https://user-images.githubusercontent.com/42147639/92478492-cf55dd00-f214-11ea-88e0-e8e9d1a33330.png)
for v5.7.2

**Comment by dziekonskik** [comment]
I would like to have this feature as well, I want to use address previously saved by the user in the dashboard but use this same handler as I have some logic there that could be reused because the shape of the data is this same and i can not because validation does not allow me to click the submit button

**Comment by sega057** [comment]
Similar behaviour I've implemented with zod validation for "Save as draft" and "Regular submittion" buttons.
You can use this trick with ref and return anything you want in resolver function.
```
export const draftResolver = zodResolver<zod.ZodType<Partial<YourPayloadType>>>(
    zod.object({
        id: zod.string({
            required_error: 'id is required',
        }),
    }),
);

export const defaultResolver = zodResolver<zod.ZodType<YourPayloadType>>(
...

// Inside a component
const isDraft = React.useRef(false);

const methods = useForm({
    resolver: (...args) => {
        if (isDraft.current) {
            return draftResolver(...args);
        }
        return defaultResolver(...args);
    },
    defaultValues,
});

const onSubmit = (draft?: boolean) => {
    isDraft.current = Boolean(draft);

    methods.handleSubmit(async (data) => {
        // any logic you want to submit the form
    })();
};
```

**Comment by NishilSB** [comment]
i found a simple solution, hence posting here 
https://codesandbox.io/s/crazy-knuth-9h4mft?file=/src/App.js

**Reply by jcready** [reply]
Unfortunately this doesn't work with the `formState.isSubmitting` or any of the other bits of `formState` that track the state of submission.

**Comment by Talha-24** [comment]
I am facing the same issue, actually I am using the same component in different routes by conditional rendering:
Problem : The Problem is, I want to disable my react-hook-form as a whole, whenever I navigate to specific route.
As you can see:
![Screenshot (903)](https://github.com/user-attachments/assets/f49731f8-9b7e-4c74-ab35-08fd7997149d)

I am using this above components for different purpose, this time, the form is filled with valid details but,,,,,,,ever time, I trying to (Using the Complete Button which is triggering by onclick) Submit, it is again giving me validation errors
